app/views/posts/index.html.erb

```diff
@@ -1,23 +1,57 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>All Posts by User</title>
+  <title>All Posts</title>
+  <!-- Include Bootstrap CSS -->
+  <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
 </head>
 <body>
-  <h1>All Posts by User</h1>
+  <h1>All Posts</h1>
   
-  <ul>
-    <% @posts.each do |post| %>
-    <li>
-      <% if post.author %>
-        <%= link_to post.title, user_post_path(user_name: post.author.name, id: post.id) %>
-      <% else %>
-        <!-- Handle the case where author is nil (e.g., display "Unknown author") -->
-        Unknown author: <%= post.title %>
+  <% @users.each do |user| %>
+    <div class="container mt-4">
+      <div class="border border-primary p-3">
+        <div class="d-flex align-items-center"> <!-- Add d-flex and align-items-center -->
+          <div class="col-md-2">
+            <!-- User's photo -->
+            <% if user.photo.present? && user.photo =~ URI::DEFAULT_PARSER.regexp[:ABS_URI] %>
+              <img src="<%= user.photo %>" alt="<%= user.name %>" class="img-fluid rounded-circle">
+            <% else %>
+              <!-- Display a default image if no valid photo URL is available -->
+              <img src="/default-user-photo.jpg" alt="Default User Photo" class="img-fluid rounded-circle">
+            <% end %>
+          </div>
+          <div class="container d-flex justify-content-between">
+            <div class='row'>
+            <h2><%= user.name %></h2>
+            </div>
+            <div class=''>
+            <p>Number of Posts: <%= user.posts.count %></p>
```

Comment:
- User details are not allowed in this page as already suggested by the previous reviewer. Please remove any code that renders user details.